flavours documentation (#2641)
redhat-developer/rhdh-operator: docs/lightspeed.md

### Configuration

#### Backend Authentication

The Lightspeed backend requires authentication configuration to connect to your LLM. Configure this through environment variables or app-config:

```yaml
spec:
  application:
    extraEnvs:
      secrets:
        - name: lightspeed-credentials
```

Ensure the secret contains the necessary authentication keys for AI service access.

#### UI Customization

The Lightspeed chat interface appears as:
- A floating action button (FAB) in the bottom-right corner
- A drawer panel that slides in from the right
- Mount points at various locations in the application

These UI elements are configured through the plugin's `pluginConfig` and can be customized in the dynamic plugins configuration.

### Features

**AI Chat Interface:**
- Contextual assistance for code development
- Documentation navigation and search
- Workflow guidance and best practices
- Natural language queries about the software catalog

### Usage

Once enabled, users can:
1. Click the Lightspeed floating button to open the chat interface
2. Ask questions about code, documentation, or workflows

### Notes

> - Lightspeed is enabled by default for all new RHDH deployments starting from version 1.10

Reviewer: ```suggestion
- Starting with version 1.10, Lightspeed is enabled by default for all RHDH deployments, including new installs and upgrades of existing instances.
```

To make it clear that is also applies to existing installs.

## Lightspeed Installation and Configuration

### Overview

Red Hat Developer Hub Lightspeed provides AI-powered developer assistance through an integrated chat interface, offering contextual help with code, documentation, and workflow guidance. The Lightspeed flavour bundles all required plugins and configurations to enable this AI-assisted experience.

### What's Included

The Lightspeed flavour (as of v1.10) consists of the following dynamic plugins:

**Lightspeed Core:**
- `backstage-plugin-lightspeed` - Frontend UI with chat interface, floating action button, and drawer components

> - `backstage-plugin-lightspeed-backend` - Backend services for AI processing

Reviewer 2: remind me @Jdubrick does this ^^ need to pivot for library model

Reviewer 3: @gabemontero No sir, the plugins on the RHDH side are identical in LCORE lib mode and server mode

Reviewer 2: Sounds good thanks

**Model Context Protocol (MCP) Tools:**
- `backstage-plugin-mcp-actions-backend` - Backend for MCP actions

> - `backstage-plugin-software-catalog-mcp-tool` - AI tools for software catalog queries

Reviewer 4: These have changed as of 1.10:

- [backstage-plugin-scaffolder-mcp-extras](https://github.com/redhat-developer/rhdh-plugin-export-overlays/pkgs/container/rhdh-plugin-export-overlays%2Fred-hat-developer-hub-backstage-plugin-scaffolder-mcp-extras)
- [backstage-plugin-techdocs-mcp-extras](https://github.com/redhat-developer/rhdh-plugin-export-overlays/pkgs/container/rhdh-plugin-export-overlays%2Fred-hat-developer-hub-backstage-plugin-techdocs-mcp-extras)
- [backstage-plugin-software-catalog-mcp-extras](https://github.com/redhat-developer/rhdh-plugin-export-overlays/pkgs/container/rhdh-plugin-export-overlays%2Fred-hat-developer-hub-backstage-plugin-software-catalog-mcp-extras)

Author: fixed

> - `backstage-plugin-techdocs-mcp-tool` - AI tools for TechDocs navigation and search

Reviewer 2: I think this is correct @johnmcollier can you double check me

Reviewer 4: These have changed since 1.10, added a comment [here](https://github.com/redhat-developer/rhdh-operator/pull/2641#discussion_r3074713366)

Reviewer 2: cool thanks @johnmcollier

### Prerequisites

To use Lightspeed, you need:
- Red Hat Developer Hub 1.10 or later

> - Access to Red Hat's AI services (configured through backend authentication)

Reviewer 2: ```suggestion
- Access to any LLM of your choosing (which you set up in the Llama Stack run.yaml configuration)
```
WDYT @Jdubrick ?

@gazarenkov - we do not restrict the array of LLMs to what Red Hat AI provides ... a customer can set up Llama Stack for example to forward to OpenAI 's endpoint (something we do regularly during dev/unit test)

Reviewer 3: Yup sounds good to me. Small nit that the config file is being renamed in my PR to `config.yaml` from `run.yaml`. But that can also be fixed in my PR depending on merge order

- (Optional) Custom MCP server configurations for extended capabilities

### Enabling Lightspeed

#### Using the Flavour (Recommended)

> Starting from version 1.10, RHDH includes Lightspeed as an **enabled-by-default** flavour. For new deployments, Lightspeed is automatically active:

Reviewer 2: ```suggestion
Starting from version 1.10, RHDH includes Lightspeed as an **enabled-by-default** flavour. For new deployments, Lightspeed is automatically active (the requisite containers are running) but inert (there is a Secret which must be updated with sufficient metadata to interact with a LLM for which you have access):
```

#### Manual Plugin Configuration

If you prefer to configure plugins manually without using the flavour, refer to the dynamic plugins ConfigMap:

```yaml
includes:
  - dynamic-plugins.default.yaml
plugins:
  - package: oci://ghcr.io/redhat-developer/rhdh-plugin-export-overlays/red-hat-developer-hub-backstage-plugin-lightspeed:bs_1.45.3__1.2.3
    disabled: false
  - package: oci://ghcr.io/redhat-developer/rhdh-plugin-export-overlays/red-hat-developer-hub-backstage-plugin-lightspeed-backend:bs_1.45.3__1.2.3
    disabled: false
  - package: oci://ghcr.io/redhat-developer/rhdh-plugin-export-overlays/backstage-plugin-mcp-actions-backend:bs_1.45.3__0.1.5
    disabled: false
  - package: oci://ghcr.io/redhat-developer/rhdh-plugin-export-overlays/red-hat-developer-hub-backstage-plugin-software-catalog-mcp-tool:bs_1.45.3__0.4.1
    disabled: false
  - package: oci://ghcr.io/redhat-developer/rhdh-plugin-export-overlays/red-hat-developer-hub-backstage-plugin-techdocs-mcp-tool:bs_1.45.3__0.3.2
    disabled: false
```

For more information about configuring dynamic plugins, please refer to the [Configuration documentation](configuration.md).

### Configuration

#### Backend Authentication

> The Lightspeed backend requires authentication configuration to connect to Red Hat's AI services. Configure this through environment variables or app-config:

Reviewer 2: ```suggestion
The Lightspeed backend requires authentication configuration to connect to your LLM. Configure this through environment variables or app-config:
```

## Lightspeed Installation and Configuration

### Overview

Red Hat Developer Hub Lightspeed provides AI-powered developer assistance through an integrated chat interface, offering contextual help with code, documentation, and workflow guidance. The Lightspeed flavour bundles all required plugins and configurations to enable this AI-assisted experience.

### What's Included

The Lightspeed flavour (as of v1.10) consists of the following dynamic plugins:

**Lightspeed Core:**
- `backstage-plugin-lightspeed` - Frontend UI with chat interface, floating action button, and drawer components
- `backstage-plugin-lightspeed-backend` - Backend services for AI processing

**Model Context Protocol (MCP) Tools:**
* [backstage-plugin-scaffolder-mcp-extras](https://github.com/redhat-developer/rhdh-plugin-export-overlays/pkgs/container/rhdh-plugin-export-overlays%2Fred-hat-developer-hub-backstage-plugin-scaffolder-mcp-extras)
* [backstage-plugin-techdocs-mcp-extras](https://github.com/redhat-developer/rhdh-plugin-export-overlays/pkgs/container/rhdh-plugin-export-overlays%2Fred-hat-developer-hub-backstage-plugin-techdocs-mcp-extras)

> * [backstage-plugin-software-catalog-mcp-extras](https://github.com/redhat-developer/rhdh-plugin-export-overlays/pkgs/container/rhdh-plugin-export-overlays%2Fred-hat-developer-hub-backstage-plugin-software-catalog-mcp-extras)

Reviewer 3: ```suggestion

```
Lets remove this, we've decided to remove MCP in the default install of Lightspeed since that part is still dev preview

### Prerequisites

To use Lightspeed, you need:
- Red Hat Developer Hub 1.10 or later

> - Access to any LLM of your choosing (which you set up in the Llama Stack run.yaml configuration)

Reviewer 3: ```suggestion
- Access to any LLM of your choosing (which you set up in the Llama Stack config.yaml configuration)
```

Reviewer 3: Will be changed in my PR as we moved to a new naming standard

> - (Optional) Custom MCP server configurations for extended capabilities

Reviewer 3: ```suggestion

```

      enabled: true
```

To disable Lightspeed:

```yaml
spec:
  flavours:
    - name: lightspeed
      enabled: false
```

Or disable all default flavours:

```yaml
spec:
  flavours: []
```

#### Manual Plugin Configuration

If you prefer to configure plugins manually without using the flavour, refer to the dynamic plugins ConfigMap:

```yaml
includes:
  - dynamic-plugins.default.yaml
plugins:
  - package: oci://ghcr.io/redhat-developer/rhdh-plugin-export-overlays/red-hat-developer-hub-backstage-plugin-lightspeed:bs_1.45.3__1.2.3
    disabled: false
  - package: oci://ghcr.io/redhat-developer/rhdh-plugin-export-overlays/red-hat-developer-hub-backstage-plugin-lightspeed-backend:bs_1.45.3__1.2.3
    disabled: false
  - package: oci://ghcr.io/redhat-developer/rhdh-plugin-export-overlays/backstage-plugin-mcp-actions-backend:bs_1.45.3__0.1.5
    disabled: false
  - package: oci://ghcr.io/redhat-developer/rhdh-plugin-export-overlays/red-hat-developer-hub-backstage-plugin-software-catalog-mcp-tool:bs_1.45.3__0.4.1
    disabled: false
  - package: oci://ghcr.io/redhat-developer/rhdh-plugin-export-overlays/red-hat-developer-hub-backstage-plugin-techdocs-mcp-tool:bs_1.45.3__0.3.2

>     disabled: false

Reviewer 3: ```suggestion

```

```

For more information about configuring dynamic plugins, please refer to the [Configuration documentation](configuration.md).

### Configuration

#### Backend Authentication

The Lightspeed backend requires authentication configuration to connect to your LLM. Configure this through environment variables or app-config:

```yaml
spec:
  application:
    extraEnvs:
      secrets:
        - name: lightspeed-credentials
```

Ensure the secret contains the necessary authentication keys for AI service access.

#### UI Customization

The Lightspeed chat interface appears as:
- A floating action button (FAB) in the bottom-right corner
- A drawer panel that slides in from the right
- Mount points at various locations in the application

These UI elements are configured through the plugin's `pluginConfig` and can be customized in the dynamic plugins configuration.

#### MCP Server Configuration

Model Context Protocol servers can be extended with custom tools. Configure additional MCP servers in your app-config:

```yaml
# Example MCP server configuration
# (Specific configuration format depends on MCP server implementation)

> ```

Reviewer 3: ```suggestion

```

### Features

**AI Chat Interface:**
- Contextual assistance for code development
- Documentation navigation and search
- Workflow guidance and best practices
- Natural language queries about the software catalog

**Software Catalog Integration:**
- Query components, APIs, and resources using natural language
- Get information about ownership, dependencies, and relationships

**TechDocs Integration:**
- Search and navigate technical documentation
- Ask questions about documentation content

> - Get contextual help based on current documentation

Reviewer 3: ```suggestion

```

### Usage

Once enabled, users can:
1. Click the Lightspeed floating button to open the chat interface
2. Ask questions about code, documentation, or workflows
3. Query the software catalog using natural language

> 4. Navigate TechDocs with AI assistance

Reviewer 3: ```suggestion

```

### Notes

- Lightspeed is enabled by default for all new RHDH deployments starting from version 1.10
- The flavour includes all necessary UI components and backend services
- MCP tools provide AI access to catalog and documentation data

> - Custom MCP servers can be added for organization-specific capabilities

Reviewer 3: ```suggestion

```
